client/src/templates/Challenges/classic/editor.tsx

```diff
@@ -68,22 +68,22 @@ import './editor.css';
 
 const MonacoEditor = Loadable(() => import('react-monaco-editor'));
 
-interface EditorProps {
+export interface EditorProps {
   attempts: number;
   canFocus: boolean;
   challengeFiles: ChallengeFiles;
   challengeType: number;
   containerRef: MutableRefObject<HTMLElement | undefined>;
-  contents: string;
+  contents?: string;
```

Comment:
```suggestion
```
The `contents` comes from challenge files, not like this.

Reply:
Do you think these 4 types can be treated the same as `contents`.

```ts
  | 'initialExt'
  | 'initialEditorContent'
  | 'ext'
  | 'dimensions'
```